File: 242-valid-anagram/valid-anagram.ts
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    for (let i = 0; i < s.length; i++){
        let hitIndex = t.indexOf(s[i]);

        if(hitIndex == -1){
            return false
        }

        t = t.slice(0, hitIndex) + t.slice(hitIndex+1)
    }

    if(t.length !== 0){
        return false
    }

    return true
    
};

File: 242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        for i in range(len(s)):
            hit_index = t.find(s[i])
            
            if hit_index == -1:
                return False
            
            t = t[:hit_index] + t[hit_index+1:]
        
        if len(t) != 0:
            return False
        
        return True
        